tests/testthat/test-preparedata-scatterplot.R

```diff
@@ -149,3 +149,152 @@ test_that("Handle y-values in multiple columns + multiple statistics",
+NET = 100, Male = 49.375, Female = 50.625, NET = 100), .Dim = 13L, .Dimnames = list(
+    c("Less than $15,000", "$15,001 to $30,000", "$30,001 to $45,000",
+    "$45,001 to $60,000", "$60,001 to $90,000", "$90,001 to $120,000",
+    "$120,001 to $150,000", "$150,001 to $200,000", "$200,001 or more",
+    "NET", "Male", "Female", "NET")), statistic = "%", basedescriptiontext = "sample size = from 707 to 800; total sample size = 800; 93 missing", basedescription = list(
+    Minimum = 707L, Maximum = 800L, Range = TRUE, Total = 800L,
+    Missing = 93L, EffectiveSampleSize = 800L, EffectiveSampleSizeProportion = 100,
+    FilteredProportion = 0), questiontypes = "PickAny", span = list(
+    rows = structure(list(c("Income", "Income", "Income", "Income",
+    "Income", "Income", "Income", "Income", "Income", "Income",
+    "Gender", "Gender", "Gender"), c("Less than $15,000", "$15,001 to $30,000",
+    "$30,001 to $45,000", "$45,001 to $60,000", "$60,001 to $90,000",
+    "$90,001 to $120,000", "$120,001 to $150,000", "$150,001 to $200,000",
+    "$200,001 or more", "NET", "Male", "Female", "NET")), class = "data.frame", .Names = c("",
+    ""), row.names = c(NA, 13L))), name = "table.BANNER", questions = c("BANNER",
+"SUMMARY")), Z2 = NULL, groups = NULL, labels = NULL, structure(c(`Less than $15,000` = -15.85213900942,
+`$15,001 to $30,000` = -11.4519011349191, `$30,001 to $45,000` = -11.7778446811784,
+`$45,001 to $60,000` = -7.2146350335478, `$60,001 to $90,000` = -4.85154432316768,
+`$90,001 to $120,000` = -9.33326808423343, `$120,001 to $150,000` = -12.918647093086,
+`$150,001 to $200,000` = -14.1409353915585, `$200,001 or more` = -14.385393051253,
+NET = 39.884207400925, Male = 11.4020968466331, Female = 12.1681291929185,
+NET = 42.4264068711928), .Dim = 13L, .Dimnames = list(c("Less than $15,000",
+"$15,001 to $30,000", "$30,001 to $45,000", "$45,001 to $60,000",
+"$60,001 to $90,000", "$90,001 to $120,000", "$120,001 to $150,000",
+"$150,001 to $200,000", "$200,001 or more", "NET", "Male", "Female",
+"NET")), statistic = "z-Statistic", basedescriptiontext = "sample size = from 707 to 800; total sample size = 800; 93 missing", basedescription = list(
+    Minimum = 707L, Maximum = 800L, Range = TRUE, Total = 800L,
+    Missing = 93L, EffectiveSampleSize = 800L, EffectiveSampleSizeProportion = 100,
+    FilteredProportion = 0), questiontypes = "PickAny", span = list(
+    rows = structure(list(c("Income", "Income", "Income", "Income",
+    "Income", "Income", "Income", "Income", "Income", "Income",
+    "Gender", "Gender", "Gender"), c("Less than $15,000", "$15,001 to $30,000",
+    "$30,001 to $45,000", "$45,001 to $60,000", "$60,001 to $90,000",
+    "$90,001 to $120,000", "$120,001 to $150,000", "$150,001 to $200,000",
+    "$200,001 or more", "NET", "Male", "Female", "NET")), class = "data.frame", .Names = c("",
+    ""), row.names = c(NA, 13L))), name = "table.BANNER.3", questions = c("BANNER",
+"SUMMARY")))
```

Comment:
This test list variable has the 7th element that is not null but `input.data.raw` usually only has 6 inputs. Was this included by mistake?

Reply:
The 7th element is added later in wiki R code (in the loop several lines after `input.data.raw` is defined). It doesn't have a name because its added as annotation data (this happens only for Scatter plots). Its intentionally added to check that the annotation data is retained in the input data (additional tests were added for the handling of annotation data in 65bde8f)